app/labor/mailchimp_bot.rb

```diff
@@ -39,6 +39,24 @@ def upsert_to_newsletter
           }
         },
       )
+
+      success = true
+    rescue Gibbon::MailChimpError => e
+      # If user was previously subscribed, set their status to "pending"
+      return resubscribe_to_newsletter if previously_subcribed?(e)
```

Comment:
We could keep of whether or not the user was previously subscribed, but for now let's just inspect the error and try a resubscribe until we're sure this works as expected.

Reply:
I looked through the MailChimp errors we've logged. It's half "Too many request" and half "Member in Compliance state". That potentially means that we will be sending all these users a MailChimp opt-in email without them actually toggling their preference from false to true. It might be best for us not to attempt to re-subscribe here.